src/edgepi/adc/edgepi_adc.py

```diff
@@ -979,6 +987,99 @@ def set_config(
                                         [None])
         self.__config(**args)
 
+    # pylint: disable=too-many-branches
+    # pylint: disable=too-many-statements
+    def batch_read_samples_adc1(
+        self,
+        data_rate: ADC1DataRate,
+        analog_in_list: Optional[list[AnalogIn]] = None,
+        differential_pairs: Optional[list[DiffMode]] = None,
+    ) -> list:
+        """
+        This function gets a data rate (to set ADC1 to), and a list of analog in and
+        differential pairs to read from. The differential pairs & analog_in elements may overlap,
+        but usecases are be uncommon.
+
+        Will reset any prior config made to the ADC. Does not work with RTD mode, and may override configs if RTD mode
+        is active.
+
+        This function only supports ADC 1, and changes the conversion mode to PULSE automatically.
+
+        TODO: if even more performance is needed, continuous mode might only need to sleep for the first conversion!
```

Comment:
Ah, I was wondering about if this might be an issue... Good to know! I guess if we want to read every pin we can't use continuous mode, but if we just wanted to read a max of 2 pins, we might be able to configure both adc1 & adc2 in parallel?

In any case, we're well above the 100hz limit that was requested originally, so I'll remove this comment for now.

https://github.com/EdgePi-Cloud/edgepi-python-sdk/pull/424/commits/ac1596eca0d0243cf6bdf2a433220cf1f5773d92

Reply:
yes that is correct. Max 2 pin can be achieved by configuring both ADCs with different ch.